include/openthread/platform/settings.h

```diff
@@ -80,10 +80,16 @@ enum
 /**
  * Performs any initialization for the settings subsystem, if necessary.
  *
- * @param[in]  aInstance The OpenThread instance structure.
+ * This function also sets the critical keys that should be stored in the secure area.
+ *
+ * Note that the memory pointed by @p aCriticalKeys MUST not be released before @p aInstance is destroyed.
+ *
+ * @param[in]  aInstance            The OpenThread instance structure.
+ * @param[in]  aCriticalKeys        A pointer to an array containing the list of critical keys.
```

Comment:
Could this pointer be `nullptr`? Maybe better document it?

Reply:
Yes, it can be. Updated the documentation.